File: pmd-cs/src/test/resources/net/sourceforge/pmd/lang/cs/cpd/testdata/specialComments.cs
// CPD-OFF
class Foo {
  void bar() {
    int a = 1 >> 2;
    a += 1;
    a++;
    a /= 3e2;
    float f = -3.1;
    f *= 2;
    bool b = ! (f == 2.0 || f >= 1.0 && f <= 2.0)
  }
// CPD-ON
}

File: pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/cpd/testdata/specialComments.java
package foo.bar.baz;
// CPD-OFF
// CPD-OFF
// another irrelevant comment
@ MyAnnotation ("ugh")
@NamedQueries({
                  @NamedQuery(
                  )})

public class Foo {// CPD-ON

    // special multiline comments
    class Foo /* CPD-OFF */{ } /* CPD-ON */
    class Foo /* CPD-OFF */{

        {something();}

    } /* CPD-ON */
}






